### PR TITLE
Feature/cvsb 17481

### PR DESCRIPTION
**Pull request by JonLazariniDD**
## Source control
Add code formatting and git hooks on commit-msg, prepush with linting, prettier, security checks 
[Feature/cvsb-17481](https://jira.dvsacloud.uk/browse/CVSB-17481)

## Checklist
- [x] Code has been tested manually
- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  "printWidth": 98,
+  "tabWidth": 2,
+  "useTabs": false,
+  "semi": true,
+  "singleQuote": true,
+  "quoteProps": "as-needed",
+  "trailingComma": "none",
+  "bracketSpacing": true,
+  "arrowParens": "always"
+}
```
Comment (alleypa) [inline .prettierrc:11]: lets remove the terminal line

File: commitlint.config.js [hunk 2]
```diff
@@ -0,0 +1 @@
+module.exports = { extends: ['@commitlint/config-conventional'] };
```
Comment (alleypa) [inline commitlint.config.js:1]: terminal line

**Comment by alex-deloitte** [comment]
This might generate lots of conflicts with tickets currently in development. Will need to limit changes before this is merged

**Comment by JonLazariniDD** [comment]
@alex-deloitte it will you are right, thanks for pointing this out.
We could get this in before merging other features then merging back features again,
All the files have been prettifyied and it should not change much once this is in for the changes to be applied ontop.
The linting/formatting will happen on commit-msg hook.
Waiting for the automation tests.

**Comment by JonLazariniDD** [comment]
full regression ran: https://jenkins.cvs.dvsacloud.uk/job/UPDATE__BRANCH/job/job_feature_test_mobile/463/DVSA_20CVS_20Mobile_20Automation/
Same threshold as develop

File: src/modules/logs/logs.actions.ts [hunk 3]
```diff
@@ -51,16 +48,15 @@ export class LoadLog implements Action {
...
 export type Types =
-  SaveLog |
-  StartSendingLogs |
-  SendLogs |
-  SendLogsSuccess |
-  SendLogsFailure |
-  PersistLog |
-  LoadLog |
-  LoadLogState;
+  | SaveLog
```
Comment (alex-deloitte) [inline src/modules/logs/logs.actions.ts:55]: Will this work? A pipe at the beginning?

Comment (JonLazariniDD) [reply src/modules/logs/logs.actions.ts:55]: yep 👍

**Pull request merged**